c++/src/capnp/compat/websocket-rpc.c++

```diff
@@ -0,0 +1,107 @@
+#include <capnp/compat/websocket-rpc.h>
+#include <kj/io.h>
+#include <capnp/serialize.h>
+
+namespace capnp {
+
+WebSocketMessageStream::WebSocketMessageStream(kj::WebSocket& socket, size_t maxMessageReceiveSize)
+  : socket(socket),
+    maxMessageReceiveSize(maxMessageReceiveSize)
+  {};
+
+kj::Promise<kj::Maybe<MessageReaderAndFds>> WebSocketMessageStream::tryReadMessage(
+    kj::ArrayPtr<kj::AutoCloseFd> fdSpace,
+    ReaderOptions options, kj::ArrayPtr<word> scratchSpace) {
+  return socket.receive(maxMessageReceiveSize)
+    .then([options, scratchSpace](auto msg) -> kj::Promise<kj::Maybe<MessageReaderAndFds>> {
+      KJ_SWITCH_ONEOF(msg) {
+        KJ_CASE_ONEOF(closeMsg, kj::WebSocket::Close) {
+          return kj::Maybe<MessageReaderAndFds>();
+        }
+        KJ_CASE_ONEOF(str, kj::String) {
+          KJ_FAIL_REQUIRE(
+            "Unexpected websocket text message; expected only binary messages."
+          );
+          break;
+        }
+        KJ_CASE_ONEOF(bytes, kj::Array<byte>) {
+          kj::ArrayPtr<byte> ptr(bytes);
+          if(reinterpret_cast<uintptr_t>(&bytes[0]) % alignof(word) == 0
```

Comment:
Use `bytes.begin()` instead of `&bytes[0]` to avoid an out-of-bounds array access exception when the array is empty.

```suggestion
          if(reinterpret_cast<uintptr_t>(bytes.begin()) % alignof(word) == 0
```